### PR TITLE
[FLINK-33758] Implement restore tests for TemporalSort node

### PR DESCRIPTION
**Pull request by jnh5y**
## What is the purpose of the change

Implement restore tests for TemporalSort node

## Verifying this change

This change added tests and can be verified as follows:

* Added restore tests for TemporalSort node which verifies the generated compiled plan with the saved compiled plan

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                                    .addSchema(
+                                            "ts STRING",
+                                            "`int` INT",
+                                            "`double` DOUBLE",
+                                            "`float` FLOAT",
+                                            "`bigdec` DECIMAL(10, 2)",
+                                            "`string` STRING",
+                                            "`name` STRING",
+                                            "`rowtime` AS TO_TIMESTAMP(`ts`)",
+                                            "WATERMARK for `rowtime` AS `rowtime` - INTERVAL '1' SECOND")
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java:89]: do we need all of the columns if we use only 3?

**Review by dawidwys — APPROVED** [review]
one minor comment, otherwise lgtm

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java (hunk 1 above)
Comment (jnh5y) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java:89]: We can certainly remove them.  I used one of the common blocks of data from another restore test.  

I think we can head one of three directions:

1. Share common data.
2. Repeat data in its entirety.
3. Repeat data in a projected form.  

There are arguments for each approach.  The existing tests have use approach 1 and I'm using 2 here. 

Which way would you like / recommend?

Comment (dawidwys) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java:89]: Does this use data from other test? You define the data here.

The more self-contained the test is, the easier it is to read it without clicking through dozens of clasess.

Comment (jnh5y) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java:89]: https://github.com/apache/flink/pull/23879/files#diff-522875e33622ba739132c4f1f2bf8ecb4140ff0142c337ead614f26ed343d033R31-R47 is more or less the same as https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupWindowAggregateTestPrograms.java#L31-L51

Comment (dawidwys) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TemporalSortTestPrograms.java:89]: Does it need to be? I see no benefit of using such a complex data for `ORDER BY`

**Pull request merged**